### PR TITLE
[private_dc] add info panel configs for scatter and timeline

### PR DESCRIPTION
**Pull request by beets**
Moved the implementation of config --> jsx to a memoized, shared component.

Tested and there are no diffs with this change.

### CODE REVIEW

File: static/js/tools/shared/info_examples.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+declare global {
+  interface Window {
+    // Stored config of sample links for the landing page.
+    infoConfig: InfoRow[];
+  }
+}
+
+function generateLinksJsx(links: InfoLink[]): JSX.Element[] {
+  if (!links) return;
```
Comment (shifucun) [inline static/js/tools/shared/info_examples.tsx:53]: return null here?

File: static/js/tools/shared/info_examples.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      <React.Fragment key={ei}>
+        <a href={link.url}>{link.text}</a>
+        {punctuation}
+      </React.Fragment>
+    );
+  });
+}
+
+function generateSubheadersJsx(subheaders: InfoSubheader[]): JSX.Element[] {
+  if (!subheaders) return;
```
Comment (shifucun) [inline static/js/tools/shared/info_examples.tsx:67]: return null here?

File: static/js/tools/shared/info_examples.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+interface InfoSubheader {
+  title: string;
+  examples: InfoLink[];
+}
+
+declare global {
+  interface Window {
+    // Stored config of sample links for the landing page.
```
Comment (shifucun) [inline static/js/tools/shared/info_examples.tsx:47]: nit: update comments of "landing page"

File: static/js/tools/shared/info_examples.tsx (hunk 1 above)
Comment (beets) [reply static/js/tools/shared/info_examples.tsx]: thanks - done.

File: static/js/tools/shared/info_examples.tsx (hunk 3 above)
Comment (beets) [reply static/js/tools/shared/info_examples.tsx]: thanks - done.

File: static/js/tools/shared/info_examples.tsx (hunk 2 above)
Comment (beets) [reply static/js/tools/shared/info_examples.tsx]: thanks - done

**Comment by beets** [comment]
thanks for the comments! updated.

File: static/js/tools/shared/info_examples.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  // be used to generate links similar to timelines.
+  subheaders: InfoSubheader[];
+}
+
+interface InfoSubheader {
+  title: string;
+  examples: InfoLink[];
+}
+
+declare global {
```
Comment (shifucun) [inline static/js/tools/shared/info_examples.tsx:45]: I wonder if there are other alternatives to the global Window solution, but this feels like a concise solution in this scenario. Please verify switching between the pages work as intended.

**Review by shifucun — APPROVED** [review]

**Comment by beets** [comment]
thanks for the review!

File: static/js/tools/shared/info_examples.tsx (hunk 4 above)
Comment (shifucun) [reply static/js/tools/shared/info_examples.tsx:45]: that's nice!

**Review by chejennifer — APPROVED** [review]
This is very nice!

**Pull request merged**